cxx-standard.cmake

```diff
@@ -64,7 +64,10 @@ macro(CHECK_MINIMAL_CXX_STANDARD STANDARD)
     if(_COMPILER_DEFAULT_CXX_STANDARD EQUAL 199711
         OR (_COMPILER_DEFAULT_CXX_STANDARD EQUAL 201103 AND _MINIMAL_CXX_STANDARD GREATER 11)
         OR (_COMPILER_DEFAULT_CXX_STANDARD EQUAL 201402 AND _MINIMAL_CXX_STANDARD GREATER 14)
-        OR (_COMPILER_DEFAULT_CXX_STANDARD EQUAL 201703 AND _MINIMAL_CXX_STANDARD GREATER 17))
+        OR (_COMPILER_DEFAULT_CXX_STANDARD EQUAL 201703 AND _MINIMAL_CXX_STANDARD GREATER 17)
+        OR (DEFINED CMAKE_CXX_STANDARD AND CMAKE_CXX_STANDARD LESS 98 AND _MINIMAL_CXX_STANDARD GREATER CMAKE_CXX_STANDARD)
+        OR (DEFINED CMAKE_CXX_STANDARD AND CMAKE_CXX_STANDARD EQUAL 98 AND _MINIMAL_CXX_STANDARD LESS CMAKE_CXX_STANDARD)
```

Comment:
If we are doing this we could also check for `CMAKE_CXX_STANDARD_REQUIRED` (i.e. the consumer C++ standard cannot be bumped) and error if there's an incompatibility here

Reply:
Done